code/game/objects/items/weapons/RCD.dm

```diff
@@ -22,13 +22,122 @@ RCD
 	var/datum/effect/effect/system/spark_spread/spark_system
 	var/matter = 0
-	var/max_matter = 100
+	var/max_matter = 160
 	var/working = 0
 	var/mode = 1
 	var/canRwall = 0
 	var/disabled = 0
 	var/airlock_type = /obj/machinery/door/airlock
 	var/advanced_airlock_setting = 1 //Set to 1 if you want more paintjobs available
+	var/sheetmultiplier	= 4			 //Controls the amount of matter added for each glass/metal sheet, triple for plasteel
+
+	var/list/conf_access = null
+	var/use_one_access = 0 //If the airlock should require ALL or only ONE of the listed accesses.
+	var/last_configurator = null
+	var/locked = 1
+
+/obj/item/weapon/rcd/verb/change_airlock_access()
+	set name = "Change Airlock Access"
+	set category = "Object"
+	set src in usr
+
+	if (!ishuman(usr))
+		return ..(usr)
+
+	var/mob/living/carbon/human/H = usr
+	if(H.getBrainLoss() >= 60)
+		return
+
+	var/t1 = text("")
+
+
+	if (last_configurator)
+		t1 += "Operator: [last_configurator]<br>"
+
+	if (locked)
+		t1 += "<a href='?src=\ref[src];login=1'>Swipe ID</a><hr>"
+	else
+		t1 += "<a href='?src=\ref[src];logout=1'>Lock Interface</a><hr>"
+
+		if(use_one_access)
+			t1 += "Restriction Type: <a href='?src=\ref[src];access=one'>At least one access required</a><br>"
+		else
+			t1 += "Restriction Type: <a href='?src=\ref[src];access=one'>All accesses required</a><br>"
+
+		t1 += "<a href='?src=\ref[src];access=all'>Remove All</a><br>"
+
+		var/accesses = ""
+		accesses += "<div align='center'><b>Access</b></div>"
+		accesses += "<table style='width:100%'>"
+		accesses += "<tr>"
+		for(var/i = 1; i <= 7; i++)
+			accesses += "<td style='width:14%'><b>[get_region_accesses_name(i)]:</b></td>"
+		accesses += "</tr><tr>"
+		for(var/i = 1; i <= 7; i++)
+			accesses += "<td style='width:14%' valign='top'>"
+			for(var/A in get_region_accesses(i))
+				if(A in conf_access)
+					accesses += "<a href='?src=\ref[src];access=[A]'><font color=\"red\">[replacetext(get_access_desc(A), " ", "&nbsp")]</font></a> "
+				else
+					accesses += "<a href='?src=\ref[src];access=[A]'>[replacetext(get_access_desc(A), " ", "&nbsp")]</a> "
+				accesses += "<br>"
+			accesses += "</td>"
+		accesses += "</tr></table>"
+		t1 += "<tt>[accesses]</tt>"
+
+	t1 += text("<p><a href='?src=\ref[];close=1'>Close</a></p>\n", src)
+
+	var/datum/browser/popup = new(usr, "airlock_electronics", "Access Control", 900, 500)
+	popup.set_content(t1)
+	popup.set_title_image(usr.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+	onclose(usr, "airlock")
+
+/obj/item/weapon/rcd/Topic(href, href_list)
+	..()
+	if (usr.stat || usr.restrained() || !ishuman(usr))
+		return
+	if (href_list["close"])
+		usr << browse(null, "window=airlock")
+		return
+
+	if (href_list["login"])
+		var/obj/item/I = usr.get_active_hand()
+		if (istype(I, /obj/item/device/pda))
+			var/obj/item/device/pda/pda = I
+			I = pda.id
+		if (I && src.check_access(I))
+			src.locked = 0
+			src.last_configurator = I:registered_name

```

Comment:
I stole this from airlock_electronics, and left it almost unchanged. If this needs to be changed it's material for another PR, one that would fix setting access in doors and keep it contained there instead of awkwardly copying it from airlock electronics.